ragstack-e2e-tests/e2e_tests/langchain_llamaindex/test_astra.py

```diff
@@ -224,13 +223,10 @@ def __init__(self, astra_ref):
         self.astra_ref = astra_ref
 
 
-def delete_collection(astra_ref):
-    db = LibAstraDB(token=astra_ref.token, api_endpoint=astra_ref.api_endpoint)
-    db.delete_collection(astra_ref.collection)
-
-
 @pytest.fixture
 def environment():
-    astra_ref = get_astra_ref()
-    yield Environment(astra_ref=astra_ref)
-    delete_collection(astra_ref)
+    handler = get_vector_database_handler()
+    handler.ensure_implements_astradb()
+    handler.before_test("astradb")
```

Comment:
Could `ensure_implements_astradb()` move into `before_test("astradb")` with `if env is "astradb": ensure_...`? 

Would be nice in fixtures to only have to remember to call `before_test(..)` and `after_test(..)`.